backend/amd64/cfg_selection.ml

```diff
@@ -128,10 +128,13 @@ let specific x : Cfg.basic_or_terminator = Basic (Op (Specific x))
 let pseudoregs_for_operation op arg res =
   match (op : Operation.t) with
   (* Two-address binary operations: arg.(0) and res.(0) must be the same *)
-  | Intop (Iadd | Isub | Imul | Iand | Ior | Ixor)
-  | Floatop ((Float32 | Float64), (Iaddf | Isubf | Imulf | Idivf))
-  | Specific Ipackf32 ->
+  | Intop (Iadd | Isub | Imul | Iand | Ior | Ixor) | Specific Ipackf32 ->
     [| res.(0); arg.(1) |], res
+  | Floatop ((Float32 | Float64), (Iaddf | Isubf | Imulf | Idivf))
+  | Specific (Ifloatarithmem (_, _, _)) ->
+    if Proc.has_three_operand_float_ops ()
+    then raise Use_default_exn
+    else [| res.(0); arg.(1) |], res
```

Comment:
Isn't this a limitation we would like
to overcome (in a future pull request)?

Reply:
By limitation, I meant the fact that
`arg.(0)` is set to `res.(0)`.

Reply:
Oh, sure - that is already happening in `emit`, but yeah it might be better to make the decision during selection